### PR TITLE
Adding uap and uapaot vertical builds

### PR DESCRIPTION
**Pull request by joperezr**
Adding uap and uapaot vertical CI runs to ensure that they don't get broken unintentionally.

cc: @weshaggard @mmitche 

FYI: @stephentoub if you see any of these two runs failing on PRs feel free to tag me so that I can take a look.

### CODE REVIEW

**Comment by joperezr** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -297,6 +297,41 @@ def buildArchConfiguration = ['Debug': 'x86',
...
+            ['Windows_NT'].each { osName ->
+                def osGroup = osGroupMap[osName]
+                def osForMachineAffinity = osName
+
+                def newJobName = "${targetGroup}_${configurationGroup.toLowerCase()}"
+
+                def newJob = job(Utilities.getFullJobName(project, newJobName, isPR)) {
+                    // On Windows we use the packer to put together everything. On *nix we use tar
+                    steps {
+                        batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd -${configurationGroup} -os:${osGroup} -buildArch:${buildArchConfiguration[configurationGroup]} -framework:${targetGroup}")
```
Comment (weshaggard) [inline netci.groovy]: Do you really want these configurations to be buildArch x86?

**Comment by mmitche** [comment]
LGTM, you should manually run the test generated job to make sure.

**Comment by joperezr** [comment]
Will do

File: netci.groovy (hunk 1 above)
Comment (joperezr) [inline netci.groovy]: nope, I noticed that they will be x86 once the jobs where generated. That should still work so I have already kicked off a couple of runs, but I will change them back to x64 before merging.

**Comment by jkotas** [comment]
I am seeing build breaks in uapaot introduced by recent changes - make sure to sync to latest master for verification.

**Comment by joperezr** [comment]
@dotnet-bot test ci please

**Comment by joperezr** [comment]
I've verified that the two new CIs are green now, so I'll merge this now.

**Pull request merged**